### PR TITLE
Add rasa x --config CONFIG

### PR DESCRIPTION
**Pull request by ArjaanBuijk**
**Proposed changes**:
- Add rasa x --config CONFIG
- See [Issue 4223](https://github.com/RasaHQ/rasa/issues/4223)

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] modified existing test for the functionality
- [ ] updated the documentation
- [x] updated the changelog
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa_nlu#code-style) for instructions)

### CODE REVIEW

**Comment by federicotdn** [comment]
IMO this is ready to be merged.

Regarding testing: an ideal test would first create an empty project with `rasa init`, then change the `config.yml` file's name to `foobar.yml` (for example), then try running `rasa x --config foobar.yml`, which should load the config file properly. This would need to be done without actually launching Rasa X. Ideally we would use the `run` fixture in `tests/cli/conftest.py` but then we wouldn't be able to mock the launching of Rasa X because `run` starts a separate process.

Also I believe we should merge this first, do a Rasa release and then update the Rasa X dependency. What I'm not sure is if this should go in Rasa 1.5 or in the next 1.4 patch.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -16,6 +16,8 @@ Added
 - Added the KeywordIntentClassifier
 - Fall back to ``InMemoryTrackerStore`` in case there is any problem with the current
   tracker store
+- Add command line argument ``rasa x --config CONFIG``, to specify path to the policy and
+  NLU pipeline configuration of your bot. (default: config.yml)
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
  NLU pipeline configuration of your bot (default: config.yml)
```

File: rasa/cli/x.py [hunk 2]
```diff
@@ -346,6 +347,14 @@ def run_in_production(args: argparse.Namespace):
     _rasa_service(args, endpoints, None, credentials_path)
 
 
+def _get_config_path(args: argparse.Namespace,) -> Optional[Text]:
+    config_path = cli_utils.get_validated_path(
+        args.config, "config", DEFAULT_CONFIG_PATH
+    )
+
+    return config_path
```
Comment (wochinge) [inline rasa/cli/x.py]: I wonder whether we should pass this `validated` config path into Rasa X so we don't need these lines of codes in Rasa X 

```python
if config_path:
        config.default_config_path = config_path
else:
        config.default_config_path = LOCAL_CONFIG_PATH
```
but could just do
```python
config.default_config_path = config_path
```

Comment (wochinge) [reply rasa/cli/x.py]: what do you think?

**Comment by wochinge** [comment]
> Also I believe we should merge this first, do a Rasa release and then update the Rasa X dependency. What I'm not sure is if this should go in Rasa 1.5 or in the next 1.4 patch.

Good question. I suggest to be strict about it. It's a new feature (though it's a small change) and hence should go into 1.5

**Comment by wochinge** [comment]
> Regarding testing: an ideal test would first create an empty project with rasa init, then change the config.yml file's name to foobar.yml (for example), then try running rasa x --config foobar.yml, which should load the config file properly. This would need to be done without actually launching Rasa X. Ideally we would use the run fixture in tests/cli/conftest.py but then we wouldn't be able to mock the launching of Rasa X because run starts a separate process.

Yeah, it's tricky since it's right at the edge between rasa / rasa-x. For me it would be okay, to not add another test in rasa and maybe just adding a unit test in rasa-x for `_configure_for_local_server`

**Comment by tmbo** [comment]
@ArjaanBuijk can you take another look at this so we can move this forward?

File: rasa/cli/x.py (hunk 2 above)
Comment (ArjaanBuijk) [reply rasa/cli/x.py]: Yes, I like that! 
I have implemented it & will show up in next push.

Should we do something similar for these ones that are defined in rasax/community/local.py:
LOCAL_DATA_DIR = "data"
LOCAL_DEFAULT_NLU_FILENAME = "nlu.md"
LOCAL_DEFAULT_STORIES_FILENAME = "stories.md"
LOCAL_DOMAIN_PATH = "domain.yml"
LOCAL_MODELS_DIR = "models"

Probably as part of a new issue...

File: CHANGELOG.rst [hunk 3]
```diff
@@ -16,6 +16,8 @@ Added
 - Added the KeywordIntentClassifier
 - Fall back to ``InMemoryTrackerStore`` in case there is any problem with the current
   tracker store
+- Add command line argument ``rasa x --config CONFIG``, to specify path to the policy and
+  NLU pipeline configuration of your bot (default: config.yml)
```
Comment (wochinge) [inline CHANGELOG.rst]:
```suggestion
  NLU pipeline configuration of your bot (default: ``config.yml``)
```

File: rasa/cli/x.py [hunk 4]
```diff
@@ -220,8 +220,9 @@ def _configure_logging(args: argparse.Namespace):
         logging.getLogger("py.warnings").setLevel(logging.ERROR)
 
 
-def is_rasa_project_setup(project_path: Text):
-    mandatory_files = [DEFAULT_CONFIG_PATH, DEFAULT_DOMAIN_PATH]
+def is_rasa_project_setup(args: argparse.Namespace, project_path: Text):
```
Comment (wochinge) [inline rasa/cli/x.py]:
```suggestion
def is_rasa_project_setup(args: argparse.Namespace, project_path: Text) -> bool:
```

File: rasa/cli/arguments/x.py [hunk 5]
```diff
@@ -1,6 +1,10 @@
 import argparse
 
-from rasa.cli.arguments.default_arguments import add_model_param, add_data_param
+from rasa.cli.arguments.default_arguments import (
+    add_model_param,
```
Comment (wochinge) [inline rasa/cli/arguments/x.py]: we try to not import functions directly, but rather the module which contains them. E.g. 

```
from rasa.cli.arguments import default_arguments

default_arguments.add_model_param(...)
```

Could you please adapt your import and the old ones? :-)

File: rasa/cli/x.py [hunk 6]
```diff
@@ -380,8 +389,10 @@ def run_locally(args: argparse.Namespace):
     rasa_x_token = generate_rasa_x_token()
     process = start_rasa_for_local_rasa_x(args, rasa_x_token=rasa_x_token)
 
+    config_path = _get_config_path(args)
+
     try:
-        local.main(args, project_path, args.data, token=rasa_x_token)
+        local.main(args, project_path, args.data, config_path, token=rasa_x_token)
```
Comment (wochinge) [inline rasa/cli/x.py]: @ArjaanBuijk  https://github.com/RasaHQ/rasa-x/pull/1537#discussion_r346417119

Comment (ArjaanBuijk) [reply rasa/cli/x.py]: All clear and modified in next commit.

File: rasa/cli/x.py [hunk 7]
```diff
@@ -380,8 +389,12 @@ def run_locally(args: argparse.Namespace):
     rasa_x_token = generate_rasa_x_token()
     process = start_rasa_for_local_rasa_x(args, rasa_x_token=rasa_x_token)
 
+    config_path = _get_config_path(args)
+
     try:
-        local.main(args, project_path, args.data, token=rasa_x_token)
+        local.main(
+            args, project_path, args.data, token=rasa_x_token, config_path=config_path
```
Comment (wochinge) [inline rasa/cli/x.py]: 🚀

**Pull request merged**